Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,714 @@
+
+namespace CompactlySupportedContinuousMap
+
+section Basics
+
+variable [TopologicalSpace β] [Zero β] [FunLike F α β] [CompactlySupportedContinuousMapClass F α β]
+
+instance instFunLike : FunLike C_c(α, β) α β where
+  coe f := f.toFun
+  coe_injective' f g h := by
+    obtain ⟨⟨_, _⟩, _⟩ := f
+    obtain ⟨⟨_, _⟩, _⟩ := g
+    congr
+
+instance instCompactlySupportedContinuousMapClass :
+    CompactlySupportedContinuousMapClass C_c(α, β) α β where
+  map_continuous f := f.continuous_toFun
+  hasCompactSupport f := f.hasCompactSupport'
+
+instance instCoeTC : CoeTC F C_c(α, β) :=
+  ⟨fun f =>
+    { toFun := f
+      continuous_toFun := map_continuous f
+      hasCompactSupport' := hasCompactSupport f }⟩
+
+@[simp]
+theorem coe_toContinuousMap (f : C_c(α, β)) : (f.toContinuousMap : α → β) = f :=
+  rfl
+
+@[ext]
+theorem ext {f g : C_c(α, β)} (h : ∀ x, f x = g x) : f = g :=
+  DFunLike.ext _ _ h
+
+@[simp]
+theorem coe_mk (f : C(α, β)) (h : HasCompactSupport f) : ⇑(⟨f, h⟩ : C_c(α, β)) = f :=
+  rfl
+
+/-- Copy of a `CompactlySupportedContinuousMap` with a new `toFun` equal to the old one. Useful
+to fix definitional equalities. -/
+protected def copy (f : C_c(α, β)) (f' : α → β) (h : f' = f) : C_c(α, β) where
+  toFun := f'
+  continuous_toFun := by
+    rw [h]
+    exact f.continuous_toFun
+  hasCompactSupport' := by
+    simp_rw [h]
+    exact f.hasCompactSupport'
+
+@[simp]
+theorem coe_copy (f : C_c(α, β)) (f' : α → β) (h : f' = f) : ⇑(f.copy f' h) = f' :=
+  rfl
+
+theorem copy_eq (f : C_c(α, β)) (f' : α → β) (h : f' = f) : f.copy f' h = f :=
+  DFunLike.ext' h
+
+theorem eq_of_empty [IsEmpty α] (f g : C_c(α, β)) : f = g :=
+  ext <| IsEmpty.elim ‹_›
+
+/-- A continuous function on a compact space has automatically compact support. -/
+@[simps]
+def ContinuousMap.liftCompactlySupported [CompactSpace α] : C(α, β) ≃ C_c(α, β) where
+  toFun f :=
+    { toFun := f
+      continuous_toFun := f.continuous
+      hasCompactSupport' := hasCompactSupport_of_compactSpace f
+        }
+  invFun f := f
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+/-- A continuous function on a compact space has automatically compact support. This is not an
+instance to avoid type class loops. -/
+lemma compactlySupportedContinuousMapClass.ofCompact {G : Type*} [FunLike G α β]
+    [ContinuousMapClass G α β] [CompactSpace α] : CompactlySupportedContinuousMapClass G α β where
+  map_continuous := map_continuous
+  hasCompactSupport := by
+    intro f
+    exact hasCompactSupport_of_compactSpace f
+
+end Basics
+
+/-! ### Algebraic structure
+
+Whenever `β` has the structore of continuous addtive monoid and a compatible topological structure,
+then `C_c(α, β)` inherits a corresponding algebraic structure. The primary exception to this is that
+`C_c(α, β)` will not have a multiplicative identity.
+-/
+
+section AlgebraicStructure
+
+variable [TopologicalSpace β] (x : α)
+
+instance instZero [Zero β] : Zero C_c(α, β) where
```

Comment:
no need to name your instances, here and below (unless there is a good reason for it, you should remove all names given to instances in this file).
```suggestion
instance [Zero β] : Zero C_c(α, β) where
```